### PR TITLE
Add more functions to helix participant

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR adds a couple new functions to HelixParticipant. It includes functions to update disk capacity, disk availability, function to enter and exit maintenance mode. 
This PR also adds test cases for those methods.

### CODE REVIEW

**Review by JingQianCloud — APPROVED** [review]
With my limited knowledge of HelixParticipant, the PR looks good to me.

What's the case that we want to change disk's capacity? Is it used to fix some wrong configuration? Do we want to increase or decrease the capacity?

**Comment by justinlin-linkedin** [comment]
> With my limited knowledge of HelixParticipant, the PR looks good to me.
> 
> What's the case that we want to change disk's capacity? Is it used to fix some wrong configuration? Do we want to increase or decrease the capacity?

There is one case we will be decreasing the disk capacity, that's when some disks are broken and we have to move replicas on those disks out. 
And in the opposite, there will be one case we want to increase the disk capacity, that's when the broken disks are fixed and we want helix to know about the fix.

**Review by SophieGuo410 — COMMENTED** [review]
Overall lgtm! Left some minor comments.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java [hunk 1]
```diff
@@ -304,6 +312,76 @@ public boolean resetPartitionState(String partitionName) {
     return result;
   }
 
+  @Override
+  public boolean resetPartitionState(List<String> partitionNames) {
```
[the remark this answers is not in the document]
Comment (Arun-LinkedIn) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java:316]: @justinlin-linkedin can correct me. I think this method might help us to send all the partitions to be reset **at once** to Helix via `helixAdmin.resetPartition(clusterName, instanceName, resourceName, partitions)`. But, may be we could deprecate `resetPartitionState(String partitionName)` API and use only this one.

**Review by Arun-LinkedIn — APPROVED** [review]

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java [hunk 2]
```diff
@@ -304,6 +312,76 @@ public boolean resetPartitionState(String partitionName) {
...
+          DataNodeConfig.DiskConfig newDiskConfig =
+              new DataNodeConfig.DiskConfig(state, diskConfig.getDiskCapacityInBytes());
+          newDiskConfig.getReplicaConfigs().putAll(diskConfig.getReplicaConfigs());
+          dataNodeConfig.getDiskConfigs().put(diskId.getMountPath(), newDiskConfig);
+        }
+      }
+
+      if (dataNodeConfigChanged) {
+        if (!dataNodeConfigSource.set(dataNodeConfig)) {
+          logger.warn("Setting disk {} state failed DataNodeConfig update");
```
[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java]: updated

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java [hunk 3]
```diff
@@ -304,6 +312,76 @@ public boolean resetPartitionState(String partitionName) {
...
+  }
+
+  @Override
+  public boolean setDisksState(List<DiskId> diskIds, HardwareState state) {
+    if (diskIds == null || diskIds.isEmpty()) {
+      throw new IllegalArgumentException("List of disk is empty when set disks state");
+    }
+    for (DiskId diskId : diskIds) {
+      if (!(diskId instanceof AmbryDisk)) {
+        throw new IllegalArgumentException("HelixParticipant only works with the AmbryDisk implementation of DiskId");
```
[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java]: updated

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java:316]: that's right, the resetPartitionState method is used by an admin request to control replica state. I don't want to change that yet.
But resetPartitionsState would change a list of partitions state in one single call. This should be faster.

**Review by SophieGuo410 — APPROVED** [review]

**Pull request merged**